snippets/core/system-text-json/csharp/WeatherForecastRequiredPropertyConverter.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class WeatherForecastRequiredPropertyConverter : JsonConverter<WeatherForecast>
+    {
+        public override WeatherForecast Read(
+            ref Utf8JsonReader reader,
+            Type type,
+            JsonSerializerOptions options)
+        {
+            // Don't pass in options when recursively calling Deserialize.
+            WeatherForecast value = JsonSerializer.Deserialize<WeatherForecast>(ref reader);
```

Comment:
Re-iterating https://github.com/dotnet/docs/pull/16225#discussion_r362137385 since it makes more sense in the samples PR:

> Looks like using `value` as parameter names isn't great in the docs. They show up colored blue which makes C# harder to read. Maybe use `result` or `forecast` (or maybe this issue can be fixed)?
> ![image](https://user-images.githubusercontent.com/6527137/71608014-c3b50100-2b32-11ea-8c47-a77e1ddc46bb.png)